packages/application-shell/src/components/project-container/project-container.tsx

```diff
@@ -0,0 +1,192 @@
+     * "
+     *   If a child component requires to be attached to the DOM tree
+     *   immediately when mounted, for example to measure a
+     *   DOM node, or uses 'autoFocus' in a descendant, add
+     *   state to Modal and only render the children when Modal
+     *   is inserted in the DOM tree.
+     * "
+     */
+    setLocaleSwitcherNode(document.getElementById(CONTAINERS.LOCALE_SWITCHER));
+  }, [setLocaleSwitcherNode]);
+
+  const projectKey = props.match.params.projectKey;
+  React.useEffect(() => {
+    // Ensure to sync the `projectKey` from the URL with localStorage.
+    if (projectKey) {
+      window.localStorage.setItem(STORAGE_KEYS.ACTIVE_PROJECT_KEY, projectKey);
+    }
+  }, [projectKey]);
+
+  // }
+
+  const hasNoProjects = props.user && props.user.projects.total === 0;
+
+  /**
+   * Given the user does not have any projects and account creation (sign up) is not yet
+   * enabled the user will be logged out.
+   *
+   * Given the user does not have project (and as a result is not part of an organization)
+   * the account application gets control over render. If any other application
+   * is requested to render a full page redirect (to have the proxy serve the request) occurs
+   * given the application is served by the proxy.
+   *    Given the application is not served by the proxy we do not perform a redirect as
+   *    otherwise a redirect loop can occur as no application is able to handle the route.
+   */
+  if (
+    hasNoProjects &&
+    props.environment.enableSignUp !== true &&
+    props.environment.servedByProxy
+  )
+    return <Redirect to={`/logout?reason=${LOGOUT_REASONS.NO_PROJECTS}`} />;
+  if (hasNoProjects && props.environment.enableSignUp)
+    return (
+      <Switch>
+        <Route path="/account" render={props.render} />
+        <Route component={RedirectToProjectCreate} />
+      </Switch>
+    );
+
+  return (
+    <ErrorBoundary pathname={props.location.pathname}>
+      <React.Suspense fallback={<ApplicationLoader />}>
+        <FetchProject
+          skip={!props.user || !props.user.defaultProjectKey}
+          projectKey={projectKey}
+        >
+          {({ isLoading: isProjectLoading, project }) => {
+            // TODO: do something if there is an `error`?
+            if (isProjectLoading) return <ApplicationLoader />;
+            if (!project) return <ProjectNotFound />;
+            if (project.suspension && project.suspension.isActive)
+              return (
+                <ProjectSuspended
+                  isTemporary={
+                    project.suspension.reason ===
+                    SUSPENSION_REASONS.TEMPORARY_MAINTENANCE
+                  }
+                />
+              );
+            if (project.expiry && project.expiry.isActive)
+              return <ProjectExpired />;
+            if (project.initialized === false) return <ProjectNotInitialized />;
+
+            return (
+              <ProjectDataLocale locales={project.languages}>
```

Comment:
This currently complains because of the types of array (PR is open in the backend)

<img width="697" alt="image" src="https://user-images.githubusercontent.com/1110551/72938238-e4a3f700-3d6a-11ea-8c4d-1f169b05618c.png">